### ISSUE TITLE
"Failed to restore image"

### ISSUE THREAD

**Issue by R3P-tech** [Nutlope/roomGPT#66]
I'm using Vercel, i got no issues deploying.
I've generated 5 images with success but now it keeps loading forever and i've got some errors in console:

Failed to load resource: the server responded with a status of 404 ()
api/generate:1     Failed to load resource: the server responded with a status of 504 ()
dream:1 Uncaught (in promise) SyntaxError: Unexpected token 'A', "An error o"... is not valid JSON

After few test, it can randomly happen that image doesn't generate and credits are used.
All requests are generated on Replicate. It just doesn't return to server

**Comment by R3P-tech** [comment]
api/generate:1     Failed to load resource: the server responded with a status of 504 ()
dream:1 Uncaught (in promise) SyntaxError: Unexpected token 'A', "An error o"... is not valid JSON
<img width="1368" alt="Capture d’écran 2023-04-13 à 13 32 57" src="https://user-images.githubusercontent.com/130335349/231675357-6f6edce9-1909-46e2-8107-6d99f6e89bf1.png">

**Comment by boss2256** [comment]
i have the same error 
info - Loaded env from /var/task/.env
![error](https://user-images.githubusercontent.com/72273305/233194437-d5f96822-e856-4b6b-bcd8-d44576456260.png)

**Comment by benderlidze** [comment]
Hey, looks like the problem is in generate.ts 
use 
generatedImage = jsonFinalResponse.output[0] as string;
instead of 
generatedImage = jsonFinalResponse.output[1] as string;

**Comment by boss2256** [comment]
> Hey, looks like the problem is in generate.ts use generatedImage = **jsonFinalResponse.output[0]** as string; instead of generatedImage = jsonFinalResponse.output[1] as string;

Hey i've already changed, the image uploaded was actually sent to replicate and upload.io but it did not return on my page.
![problem](https://user-images.githubusercontent.com/72273305/233450112-feb08fd8-47c1-4c7c-beb7-d942014a9462.png)
![Uploading problem.png…]()
![generate ts](https://user-images.githubusercontent.com/72273305/233450328-6cd32fc2-b062-4d27-8546-32bf30b3dea5.png)
![replicaete](https://user-images.githubusercontent.com/72273305/233450351-bfbf576f-09b6-4bc7-8696-ef8d978ce2ad.png)

**Comment by boss2256** [comment]
it keeps on loading forever, and it did not appear on my dashboard

**Comment by goyalpramod** [comment]
@R3P-tech @boss2256  Were you able to fix the issue?

**Comment by fewjative** [comment]
@boss2256 @goyalpramod @R3P-tech 

TLDR: You need to upgrade to vercel pro plan.

If you're getting a 504 error, it's likely because the generate API took longer than ten seconds. If you look at the last image in the first post, the right pane shows the duration as 10.01 s and limit is 10 seconds. Aka it took too long. If you deploy locally, there is NO time limit. However, when you are on production, you have time limits. If you are on **hobby**, serverless functions have a max run time of 10 seconds. On **pro**, they have a runtime max of 60.

**Comment by R3P-tech** [comment]
@fewjative Thanks, did you try with the new model from pull request #67 ?
Vercel pro solved the problem and works with the old model, but with the new model and vercel pro images are not generated.

**Comment by goyalpramod** [comment]
@fewjative thank you for the explanation, it helped tremendously.
